include/bluetooth/services/ddfs.h

```diff
@@ -0,0 +1,266 @@
+
+/** @brief UUID of the Elevation Measurement Characteristic. **/
+#define BT_UUID_DDFS_ELEVATION_MEAS_VAL \
+	BT_UUID_128_ENCODE(0x21490003, 0x494a, 0x4573, 0x98af, 0xf126af76f490)
+
+/** @brief UUID of the Feature Characteristic. **/
+#define BT_UUID_DDFS_FEATURE_VAL \
+	BT_UUID_128_ENCODE(0x21490004, 0x494a, 0x4573, 0x98af, 0xf126af76f490)
+
+/** @brief UUID of the Control Point Characteristic. **/
+#define BT_UUID_DDFS_CTRL_POINT_VAL \
+	BT_UUID_128_ENCODE(0x21490005, 0x494a, 0x4573, 0x98af, 0xf126af76f490)
+
+#define BT_UUID_DDFS                 BT_UUID_DECLARE_128(BT_UUID_DDFS_VAL)
+#define BT_UUID_DDFS_DISTANCE_MEAS   BT_UUID_DECLARE_128(BT_UUID_DDFS_DISTANCE_MEAS_VAL)
+#define BT_UUID_DFFS_AZIMUTH_MEAS    BT_UUID_DECLARE_128(BT_UUID_DDFS_AZIMUTH_MEAS_VAL)
+#define BT_UUID_DFFS_ELEVATION_MEAS  BT_UUID_DECLARE_128(BT_UUID_DDFS_ELEVATION_MEAS_VAL)
+#define BT_UUID_DDFS_FEATURE         BT_UUID_DECLARE_128(BT_UUID_DDFS_FEATURE_VAL)
+#define BT_UUID_DFFS_CTRL_POINT      BT_UUID_DECLARE_128(BT_UUID_DDFS_CTRL_POINT_VAL)
+
+/** Ranging mode */
+enum bt_ddfs_dm_ranging_mode {
+	/* Round trip timing */
+	BT_DDFS_DM_RANGING_MODE_RTT,
+
+	/* Multi-carrier phase difference */
+	BT_DDFS_DM_RANGING_MODE_MCPD,
+};
+
+enum bt_ddfs_quality {
+	/* Good measurement quality */
+	BT_DDFS_QUALITY_OK,
+
+	/* Poor measurement quality */
+	BT_DDFS_QUALITY_POOR,
+
+	/* Measurement not for use */
+	BT_DDFS_QUALITY_DO_NOT_USE,
+
+	/* Measurement quality not specified */
+	BT_DDFS_QUALITY_NONE,
+};
+
+/** Direction and Distance Service notification events. */
+enum bt_ddfs_evt {
+	/** Distance Measurement notification enable. */
+	BT_DDFS_EVT_DM_NOTIFY_ENABLE,
+
+	/** Distance Measurement notification disable. */
+	BT_DDFS_EVT_DM_NOTIFY_DISABLE,
+
+	/** Azimuth Measurement notification enable. */
+	BT_DDFS_EVT_AM_NOTIFY_ENABLE,
+
+	/** Azimuth Measurement notification disable. */
+	BT_DDFS_EVT_AM_NOTIFY_DISABLE,
+
+	/** Elevation Measurement notification enable. */
+	BT_DDFS_EVT_EM_NOTIFY_ENABLE,
+
+	/** Elevation Measurement notification disable. */
+	BT_DDFS_EVT_EM_NOTIFY_DISABLE,
+};
+
+/** @brief Bitmask set of supported features. */
+struct bt_ddfs_features {
+	/** Distance Measurement RTT ranging mode supported bit. */
+	uint8_t ranging_mode_rtt : 1;
+
+	/** Distance Measurement MCPD ranging mode supported bit. */
+	uint8_t ranging_mode_mcpd : 1;
+};
+
+/** @brief Distance Measurement configuration parameters. */
+struct bt_ddfs_dm_config {
+	/** Ranging mode */
+	enum bt_ddfs_dm_ranging_mode mode;
+};
+
+/** @brief Pointers to the callback functions for service events. */
+struct bt_ddfs_cb {
+	/** @brief Indicates that the ranging mode for Distance Measurement is set.
+	 *
+	 * @param mode New ranging mode for distanance measurement.
+	 * @retval 0 If the operation was successful.
+	 *         Otherwise, a negative error code is returned.
+	 */
+	int (*cp_dm_ranging_mode_set)(uint8_t mode);
```

Comment:
```suggestion
	int (*dm_ranging_mode_set)(uint8_t mode);
```

I think we can remove **cp_** prefix. After all, the API user does not have to know if the ranging mode is set via Control Point or other GATT mechanism (abstraction)